compiler/src/dotty/tools/dotc/core/OrderingConstraint.scala

```diff
@@ -548,7 +548,7 @@ class OrderingConstraint(private val boundsMap: ParamBounds,
 
   /** The uninstantiated typevars of this constraint */
   def uninstVars: collection.Seq[TypeVar] = {
-    if (myUninstVars == null) {
+    if (myUninstVars == null || myUninstVars.exists(_.inst.exists)) {
```

Comment:
Shouldn't this use `instanceOpt` instead of `inst`? Otherwise this will still return instantiated type variables whose `inst` field has not yet been set.

Reply:
I wonder if it wouldn't be simpler and more efficient to clear the cache everytime we instantiate a typevar instead.

Reply:
Anything in the constraint is already taken into account in `uninstVars`. After all, constraints are immutable. So we only need to make sure there are no instantiations not mirrored in the constraint. We can't clear the cache when we instantiate a typevar because there is no 1-1 mapping between constraints and typevars. So there could me multiple constraints that become invalid.